tests/enlyze/test_validators.py

```diff
@@ -0,0 +1,80 @@
+from contextlib import nullcontext as does_not_raise
+from datetime import datetime, timedelta
+from uuid import UUID
+
+import pytest
+from hypothesis import given
+from hypothesis import strategies as st
+
+from enlyze.errors import EnlyzeError, ResamplingValidationError
+from enlyze.models import Variable, VariableWithResamplingMethod
+from enlyze.validators import (
+    validate_resampling_interval,
+    validate_timeseries_arguments,
+)
+from tests.conftest import (
+    datetime_before_today_strategy,
+    datetime_today_until_now_strategy,
+)
+
+variable_strategy = st.one_of(
+    st.builds(Variable), st.builds(VariableWithResamplingMethod)
+)
+
+
+class TestValidateTimeseriesArguments:
+    @given(
+        start=datetime_before_today_strategy,
+        end=datetime_today_until_now_strategy,
+        variable=variable_strategy,
+    )
+    def test_validate_timeseries_arguments(self, start, end, variable):
+        start, end, appliance_uuid = validate_timeseries_arguments(
+            start, end, [variable]
+        )
+        assert start
+        assert end
+        assert UUID(appliance_uuid)
+
+    @given(variable=variable_strategy)
+    def test_validate_start_must_be_earlier_than_end(self, variable):
+        start = datetime.now() + timedelta(days=1)
+        end = datetime.now()
+        with pytest.raises(EnlyzeError):
```

Comment:
Please specify the expected exception more precisely

Reply:
That's too pedantic, are we going to specify the exact condition of matching for every test? That's misguided IMO.

Reply:
> If we want to test some specific validation in some hypothetical code path that might raise the same error type on 5 different conditions, how can we be sure that we have crafted the test such that we are actually testing the condition we have intended?

Most of the time we shouldn't care, in the tests you've written you use `match` to assert on the message, what I suggested elsewhere (i.e. Either add a property to the error class or add new error classes) is just to make this notion more structural and less string-based. However, in most cases we shouldn't try to disambiguate every error path explicitly because we'd end up with an overengineered error hierarchy. Reviewing the tests should be enough. It's on a case-by-case basis and IMO in these cases, it doesn't matter.